odc/stats/tasks.py

```diff
@@ -187,53 +229,92 @@ def _find_dss(
         A string joined by `+` implies intersect (filtered and then groupby) against time
         return a generator of datasets
         """
+        # pylint:disable=too-many-locals
         if dataset_filter is None:
             dataset_filter = {}
 
         # query by a list of products is not a "officially" supported feature
         # but it is embedded in the code everywhere
         # mark it for ref
 
-        def sanitize_products_str(products_str):
-            """
-            split a string composed by product names into a list of product names
-            e.g., ga_ls8-ga_ls7 -> [ga_ls8, ga_ls7]
-            rules:
-            1. Any separator (`+/-`) at the start or the end is disregarded
-            2. Multiple same separators between the product names are treated as one
-            3. Multiple different separators between the product names is respected by left-right order
-            e.g., ga_ls8+-ga_ls7 -> separator is `+` as `+` proceeds `-` from left to right
-            """
-            product_list = re.split(r"[\+-]{1,}", products_str)
-            product_list = list(filter(None, product_list))
-            group_size = len(re.findall(r"\w{1,}\+[-\+]{0,}\w{1,}", products_str))
-            return product_list, group_size
-
         product_list, group_size = sanitize_products_str(products)
 
-        query.update(dict(product=product_list, **dataset_filter))
-        dss = ordered_dss(
-            dc,
-            freq="y",
-            key=lambda ds: (
-                (ds.center_time, ds.metadata.region_code)
-                if hasattr(ds.metadata, "region_code")
-                else (ds.center_time,)
-            ),
-            **query,
-        )
+        indexed_products = []
+        non_indexed_products = []
+        for p, indexed in product_list:
+            if indexed:
+                indexed_products += [p]
+            else:
+                non_indexed_products += [p]
+
```

Comment:
I would use `arr.append(p)` here rather than `arr += [p]`.   Probably compiles to the same byte code, but I think it's easier to read.

Reply:
It's only my preference to type less. :) also, `+=` is more friendly if appending all the items in another list. I'd agree with `append` is easier to read though.